woocommerce/woocommerce-ios
# Order Creation: Revert reusable Add Product view

Part of: #5847

## Description

This reverts the changes in #5870, which made `AddProductToOrder` a reusable view for the Order Creation flow.

Instead of trying to reuse that entire view for both product and product variation lists, we'll make the infinite scroll list a reusable component and add a separate `AddProductVariationToOrder` view for showing a product variation list. (Those changes will come in another PR.)

Discussion in Slack: p1642070509055800-slack-C02KUCFCSFP

## Changes

* Reverts the commits from #5870.
* Updates `AddProductVariationToOrderViewModel` so it is a separate view model (no longer linked to the removed `AddProductToOrderViewModelProtocol`). This view model will be used for a new `AddProductVariationToOrder` view added in a future PR.
* Moves the ghost rows back to the individual view models (they were part of the protocol in the reverted changes) and simplifies how those ghost rows are instantiated.

## Testing

Confirm that the Add Product screen continues to work as expected:

1. Build and run the app in debug mode (to ensure the feature flag is enabled).
2. Make sure Order Creation is enabled under Settings > Experimental Features.
3. Go to the Orders tab, tap the + button, and select "Create order."
4. On the New Order screen, tap the "Add product" button.
5. On the Add Product screen, confirm the product list loads with all the expected details for each product.
6. Tap to select a product.
7. Confirm the selected product appears in the new order.

## Submitter Checklist

Update release notes:
